crates/cli/src/subcommands/call.rs

```diff
@@ -68,12 +66,37 @@ pub async fn exec(mut config: Config, args: &ArgMatches) -> Result<(), Error> {
     let builder = reqwest::Client::new().post(format!(
         "{}/database/call/{}/{}",
         config.get_host_url(server)?,
-        address,
+        address.clone(),
         reducer_name
     ));
     let auth_header = get_auth_header_only(&mut config, anon_identity, as_identity, server).await;
     let builder = add_auth_header_opt(builder, &auth_header);
+    let describe_reducer = util::describe_reducer(
+        &mut config,
+        address.clone(),
+        server.map(|x| x.to_string()),
+        reducer_name.clone(),
+        anon_identity,
+        as_identity.cloned(),
+    )
+    .await?;
+
+    // String quote any arguments that should be quoted
+    let arguments: Vec<String> = arguments
+        .filter(|args| args.len() == describe_reducer.schema.elements.len())
+        .unwrap_or_default()
+        .collect::<Vec<_>>()
+        .iter()
+        .zip(describe_reducer.schema.elements.iter())
+        .map(|(argument, element)| match &element.algebraic_type {
+            AlgebraicType::Builtin(BuiltinType::String) if !argument.starts_with('\"') || !argument.ends_with('\"') => {
+                format!("\"{}\"", argument)
+            }
+            _ => argument.to_string(),
+        })
```

Comment:
@gefjon Any recommendations for making this more readable?

Reply:
All told, I think this should be:

```rust
let arguments = arguments.unwrap_or_default();
let expected_count = describe_reducer.schema.elements.len();
anyhow::ensure!(
    arguments.len() == expected_count
    "Expected {} arguments but found {}",
    expected_count, arguments.len(),
);

let arguments = arguments.zip(describe_reducer.schema.elements.iter())
    .map(|(argument, element)| match &element.algebraic_type {
        AlgebraicType::Builtin(BuiltinType::String) if !argument.starts_with('\"') || !argument.ends_with('\"') => {
            format!("\"{}\"", argument)
        }
        _ => argument.to_string(),
    })
    .collect();
```

Reply:
I've ended up with this:

```rust
    // String quote any arguments that should be quoted
    let arguments = arguments.unwrap_or_default().zip(describe_reducer.schema.elements.iter()).map(|(argument, element)| match &element.algebraic_type {
        AlgebraicType::Builtin(BuiltinType::String) if !argument.starts_with('\"') || !argument.ends_with('\"') => { format!("\"{}\"", argument) }
        _ => argument.to_string(),
    }).collect::<Vec<_>>();
```

Reply:
Linter will (or should) reject that unless you add some line breaks, but it looks right.